aws/resource_aws_s3_bucket.go

```diff
@@ -708,6 +708,15 @@ func resourceAwsS3BucketRead(d *schema.ResourceData, meta interface{}) error {
 		})
 	})
 	ws := wsResponse.(*s3.GetBucketWebsiteOutput)
+
+	if err != nil {
+		if !isAWSErr(err, "NotImplemented", "") && !isAWSErr(err, "NoSuchWebsiteConfiguration", "") {
+			return err
+		}
+		log.Printf("[WARN] S3 bucket: %s, website configuration not supported by storage server.", d.Id())
+		return nil
```

Comment:
Returning here will prevent the rest of the Read function from working (versioning, acceleration, request payer, logging, lifecycle rules, etc.)

Reply:
The line shouldn't be present at all 😄

Reply:
surely if we remove it all together then the error object will be passed back and cause an exception?

Reply:
What error are you seeing? I briefly looked at the code below this and it seemed fine.

Reply:
Oops, I thought I was returning from the 'if' rather than the function.
I think rather than this I probably need to reset the error object to nil to prevent it being caught later on. 

so rather than `return nil`
we should have `err = nil`

Is that correct?

Reply:
You could do that, but its unnecessary. The line does not need to be there at all. The function will continue just fine since line 721 already does the right thing and `err` is overwritten at line 774.